Cargo.toml

```diff
@@ -15,7 +15,7 @@ serde = { version = "1.0", features = ["derive"] }
 libp2p-connection-limits = "0.5.0"
 serde_json = "1.0"
 sha2 = "0.10.6"
-tonic = "0.12.3"
+tonic = { version = "0.12.3", features = ["gzip"] }
```

Comment:
I think there were some issues upgrading tonic-build. Worth a shot if build passes, but if it requires a bunch of patches, probably not worth it.

Reply:
the automated tests pass. is there something more to check?

Reply:
It requires a fair number of patches but still would be worth picking up at some point.